src/cache.rs

```diff
@@ -331,22 +361,43 @@ impl Cache {
         &self,
         file_id: FileId,
         format: InputFormat,
-    ) -> Result<RichTerm, ParseError> {
+    ) -> Result<RichTerm, (Option<RichTerm>, ParseErrors)> {
```

Comment:
I tried a few different signatures and picked this one since it was least invasive.
I'll change it back to `Result<(RichTerm, ParseErrors), ParseError>`.

Reply:
Well, unless you have experienced that it is much more invasive, in which case that would be a valid reason not to. But it seems that the change would be limited to those few caches functions, and wouldn't bubble up that much, or would they?

Reply:
Yeah, changing it is a good idea.